File: R/657-robot-return-to-origin/main.go
package main

import "fmt"

func judgeCircle(moves string) bool {
	buf := make([]int, 2)
	for i := 0; i < len(moves); i++ {
		if moves[i] == 'R' {
			buf[0]++
		}
		if moves[i] == 'L' {
			buf[0]--
		}
		if moves[i] == 'U' {
			buf[1]++
		}
		if moves[i] == 'D' {
			buf[1]--
		}
	}
	return buf[0] == 0 && buf[1] == 0
}

func main() {
	fmt.Println("a")
}

File: R/657-robot-return-to-origin/main.rs
impl Solution {
  pub fn judge_circle(moves: String) -> bool {
    moves
      .as_bytes()
      .iter()
      .fold((true, 0, 0), |(ans, mut lr, mut ud), b| {
        lr += match b {
          b'L' => 1,
          b'R' => -1,
          _ => 0,
        };
        ud += match b {
          b'U' => 1,
          b'D' => -1,
          _ => 0,
        };
        (lr == 0 && ud == 0, lr, ud)
      })
      .0
  }
}
